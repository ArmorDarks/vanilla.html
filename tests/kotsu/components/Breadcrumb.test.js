/* eslint-env jest */

import cloneDeep from 'lodash/cloneDeep'
import { renderString } from '../../utils/nunjucks'

const render = (template, context = mockContext, parse) => renderString(template, context, parse)
const mockContext = {
  PAGE: { locale: 'en-US' },
  SITE: {
    locales: { 'en-US': { url: '/' } },
    __matter: {
      index: {
        props: { url: '/', breadcrumbTitle: 'Home' }
      }
    }
  }
}

describe('Nunjucks component `Breadcrumb()`', () => {
  it('should print proper breadcrumb', () => {
    const testContext = cloneDeep(mockContext)
    testContext.PAGE.breadcrumb = [
      'section',
      'inner-2'
    ]
    testContext.SITE.__matter.section = {
      props: { url: '/section-url', breadcrumbTitle: 'Section title' },
      'inner-1': {
        props: { url: '/section/inner-1-url', breadcrumbTitle: 'Inner 1 title' }
      },
      'inner-2': {
        props: { url: '/section/inner-2-url', breadcrumbTitle: 'Inner 2 title' }
      }
    }

    expect(render(`
      {% from '_components/_Breadcrumb.nj' import Breadcrumb with context %}
      {{ Breadcrumb() }}
    `, testContext)).toMatchSnapshot()
  })
})
