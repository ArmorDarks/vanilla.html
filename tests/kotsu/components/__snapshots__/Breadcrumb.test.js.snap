// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nunjucks component \`Breadcrumb()\` should print proper breadcrumb 1`] = `
"
      
      
<ol class='o-breadcrumb' typeof='BreadcrumbList'>

  <li class='o-breadcrumb__item o-breadcrumb__item--root' property='itemListElement' typeof='ListItem'>
    <a class=' ' href='/' property='item' typeof='WebPage'>
      <span property='name'>Home</span>
    </a>
    <meta property='position' content='1'>
  </li>

  
  

  
    
    
    
    

    
      

      

      <li class='o-breadcrumb__item' property='itemListElement' typeof='ListItem'>
        <a class='' href='/section-url' property='item' typeof='WebPage'>
          <span property='name'>Section title</span>
        </a>
        <meta property='position' content='2'>
      </li>

      
    
  
    
    
    
    

    
      

      

      <li class='o-breadcrumb__item' property='itemListElement' typeof='ListItem'>
        <a class='' href='/section/inner-2-url' property='item' typeof='WebPage'>
          <span property='name'>Inner 2 title</span>
        </a>
        <meta property='position' content='3'>
      </li>

      
    
  
    
    
    
    

    
      

      

      <li class='o-breadcrumb__item'>Deep inner title</li>

      
    
  

</ol>

    "
`;

exports[`Nunjucks component \`Breadcrumb()\` should print proper breadcrumb with excluded segment 1`] = `
"
      
      
<ol class='o-breadcrumb' typeof='BreadcrumbList'>

  <li class='o-breadcrumb__item o-breadcrumb__item--root' property='itemListElement' typeof='ListItem'>
    <a class=' ' href='/' property='item' typeof='WebPage'>
      <span property='name'>Home</span>
    </a>
    <meta property='position' content='1'>
  </li>

  
  

  
    
    
    
    

    
  
    
    
    
    

    
      

      

      <li class='o-breadcrumb__item' property='itemListElement' typeof='ListItem'>
        <a class='' href='/section/inner-2-url' property='item' typeof='WebPage'>
          <span property='name'>Inner 2 title</span>
        </a>
        <meta property='position' content='2'>
      </li>

      
    
  
    
    
    
    

    
      

      

      <li class='o-breadcrumb__item'>Deep inner title</li>

      
    
  

</ol>

    "
`;
